tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -397,11 +423,22 @@ def main():
     args = parser.parse_args()
 
     def get_all_test_cases():
-        for cipher, sig_alg, named_group, server, client in \
-            itertools.product(CIPHER_SUITE_IANA_VALUE.keys(), SIG_ALG_IANA_VALUE.keys(),
-                              NAMED_GROUP_IANA_VALUE.keys(), SERVER_CLASSES.keys(),
+        def cycle_zip(*iters):
+            max_len = max([len(i) for i in iters])
+            cycle_iters = zip(*[itertools.cycle(i) for i in iters])
+            for _, c in zip(range(max_len), cycle_iters):
+                yield c
+
+        for combine_fields, server, client in \
+            itertools.product(list(cycle_zip(CIPHER_SUITE_IANA_VALUE.keys(),
```

Comment:
Why add the function cycle_zip?

Reply:
ciphersuites, name_groups and signature algorithms are independently. So we do not need test all combinations.

Reply:
How should we decide which combination is better? and how to tell whether the combination is enough?

Reply:
Removed